### PR TITLE
add pyqt (and friends) since it was removed from homebrew-core

### PR DESCRIPTION
**Pull request by wjwwood**
Originally seen:

https://github.com/ros-infrastructure/rosdep/issues/504#issuecomment-291978495

@scpeters @mikepurvis FYI

### CODE REVIEW

**Comment by scpeters** [comment]
It's also located in the following repo:

* https://github.com/Homebrew/homebrew-boneyard/pull/243/files

**Comment by wjwwood** [comment]
Except not any more :rofl::

https://github.com/Homebrew/homebrew-boneyard/pull/314

**Comment by scpeters** [comment]
Good point. https://github.com/Homebrew/homebrew-boneyard/pull/314 says that "Homebrew 1.1.12 provides the commands necessary to show old versions of formulae". So I tried that out:

~~~
$ brew info pyqt
Error: No available formula with the name "pyqt" 
==> Searching for a previously deleted formula...
pyqt was deleted from homebrew/core in commit 2a6d326:
  pyqt: migrate to boneyard (https://github.com/Homebrew/homebrew-core/issues/6817)
  * pyqt: migrate to boneyard
  * gnuradio: remove pyqt, build docs by default.

To show the formula before removal run:
  git -C "$(brew --repo homebrew/core)" show 2a6d326^:Formula/pyqt.rb

If you still use this formula consider creating your own tap:
  http://docs.brew.sh/How-to-Create-and-Maintain-a-Tap.html
~~~

This just shows the old version of the formula, so we still need to host it somewhere. So this pull request sounds good.

**Comment by wjwwood** [comment]
@scpeters I agree. I added a short preface to the formula that talks about where it came from and why.

**Comment by scpeters** [comment]
It might need some testing though. Does this use qt4 or qt5? It depends on `qt`, which now points to `qt5`.

**Comment by scpeters** [comment]
https://github.com/Homebrew/homebrew-core/blob/master/Aliases/qt

**Comment by scpeters** [comment]
There is a qt4 formula in another popular tap, but I also have one in `osrf/simulation` with webkit explicitly disabled, since it is out of date and a security risk:

* https://github.com/osrf/homebrew-simulation/blob/master/qt4-no-webkit.rb

I'm guessing those bottles won't work anymore too.

**Comment by wjwwood** [comment]
Grrr... I guess we need to update all of the ROS rosdep rules for qt to point to `qt4` and not `qt` now too. Seems like a lot of busy work for nothing...

**Comment by wjwwood** [comment]
Looks like I'll need to do the same for `pyside`, `shiboken`, and `libqglviewer` too.

**Comment by scpeters** [comment]
Also, we are ordering a new mac mini, so we will be able to make sierra bottles soon at `osrf/simulation`

**Comment by wjwwood** [comment]
@scpeters I don't have any experience with brew-bot, any idea what I need to do in order to get it to use `qt-no-webkit` from `osrf/simulation`? See a80aebc, a8378ad, and 01e7030.

**Comment by scpeters** [comment]
I think it's better to include the tap name in the dependency: `osrf/simulation/qt4-no-webkit`, so maybe revert 01e7030a3801b7f160a538f3cbd3b0adbcea322a?

**Comment by wjwwood** [comment]
Yeah, that's what I thought but it failed too with the same issue. It also installs fine locally with at as the depend, the brew bit still fails on Travis for some reason.

**Comment by mikepurvis** [comment]
How realistic is it to try to continue to support ROS Indigo now that Homebrew has dropped upstream support for qt4?

My understanding is that Kinetic+ targets qt5; if so, maybe it makes sense to focus efforts there?

**Comment by wjwwood** [comment]
> My understanding is that Kinetic+ targets qt5; if so, maybe it makes sense to focus efforts there?

Indeed, I'm about this 👌 close to giving up on this, but I thought if all I had to do was take the output from those reproduction commands, put them in a file, remove the bottle info, and point them at the qt4 that gazebo uses, then I'd do that. At least then people might have a chance of moving forward without their own tap.

Realistically though, we probably should just drop anything that needs qt4 for versions of ROS that support qt5 fully, like Kinetic.

**Comment by scpeters** [comment]
I just tried to reproduce the travis error in a VM, and I think we would have to rename our copy of `pyqt` to distinguish it from the current version of `pyqt` in homebrew-core that uses qt5. I think the current convention is `pyqt@4.11`.

So yeah, we should probably just drop qt4 support for indigo.

**Comment by wjwwood** [comment]
Omg they just renamed `qt5` -> `qt` and `pyqt5` -> `pyqt`, like since I started working on this. That's so frustrating.

**Comment by scpeters** [comment]
They were already aliases of each other; this just changed which was the `.rb` file and which was the alias:

* https://github.com/Homebrew/homebrew-core/commit/b92c519bbd3cb869b4861017210e07fc42a0cc5f

**Comment by wjwwood** [comment]
Yeah, but before today you would get:

```

Error: No available formula with the name "pyqt" 
==> Searching for a previously deleted formula...
pyqt was deleted from homebrew/core in commit 2a6d326:
  pyqt: migrate to boneyard (https://github.com/Homebrew/homebrew-core/issues/6817)
  * pyqt: migrate to boneyard
  * gnuradio: remove pyqt, build docs by default.

To show the formula before removal run:
  git -C "$(brew --repo homebrew/core)" show 2a6d326^:Formula/pyqt.rb

If you still use this formula consider creating your own tap:
  http://docs.brew.sh/How-to-Create-and-Maintain-a-Tap.html
```

Now you can't even get the command to recover `pyqt`.

File: pyqt@4.11.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# NOTICE: this formula was removed from Homebrew core:
+#
+#   https://github.com/Homebrew/homebrew-core/pull/6817
+#
+# We've resurrected it here because ROS Indigo still needs it.
+# We used this command to get the most recent content:
+#
+#   git -C "$(brew --repo homebrew/core)" show 2a6d326^:Formula/pyqt.rb
+
+class Pyqt < Formula
```
Comment (scpeters) [inline pyqt@4.11.rb]: I think the standard practice would be to change the class name to PyqtAT411 ([example](https://github.com/Homebrew/homebrew-core/blob/master/Formula/qt@5.5.rb#L3))

Comment (scpeters) [inline pyqt@4.11.rb]: it seems like poor form to not have a separator between major and minor version numbers, but it's what they seem to do

Comment (wjwwood) [inline pyqt@4.11.rb]: Thanks, I'll try that. Just poking it a little anytime I have a spare cycle.

**Comment by wjwwood** [comment]
I still have no idea why we're getting `Error: No available formula with the name "qt4-no-webkit"` still. I tap it in the travis file, but I'm guessing `brew test-bot` is working in an isolated environment. In previous builds I thought I saw it tapping `osrf/simulation` during the testing, but the latest build doesn't look like it is.

**Comment by scpeters** [comment]
I just remembered that the `qt4-no-webkit` formula has a hard cap on supporting osx 10.11:

* https://github.com/osrf/homebrew-simulation/blob/master/qt4-no-webkit.rb#L34

but we could relax that if users demand support for sierra. I don't think that's causing the travis failures though because it's also failing on the 10.11 travis build.

**Comment by wjwwood** [comment]
Ok, I don't think this path is going to work out. I'm going to close this, and let someone else resurrect it if they think it is worthwhile.

**Pull request closed without merging**